hardware/raspberrypi/boot_diagnostics.md

```diff
@@ -0,0 +1,28 @@
+## Boot Diagnostics on the Raspberry Pi 4
+
+The bootloader on the Raspberry Pi 4 is stored in EEPROM, and as well as providing boot services, is also capable of providing diagnostic information on an attached HDMI display. To display this information, remove the SD card from the Raspberry Pi 4, and reboot. A diagnostic display similar to below should appear on the attached display.
+
+This diagnostics page will also appear if the bootloader is unable to boot from an inserted SD card, or is unable to network boot; for example, if there is no bootable image on the card, or it is defective, or the network boot parameters are incorrect.
+
+This information screen is displayed only after a reboot caused by a power cycle, not a software instigated reboot.
```

Comment:
It would be clearer if this line said something like this instead:

'In order to display the diagnostics screen, you must power down the Raspberrpi Pi, then reapply power. If you simply reboot the device, the screen will not be displayed.

Reply:
I suspect both @andrum99 and @JamesH65 have misinterpreted the original intention.

I _think_ that @timg236 meant that if the diagnostics page is being displayed, the only way to trigger a reboot is by manually power-cycling (i.e. unplugging and then re-plugging the power cable) ?

Reply:
@lurch Correct. In the current release (soon to be default) once the HDMI diagnostics page is displayed the bootloader has stopped so inserting an sd-card won't do anything without a power cycle. That behaviour is consistent with earlier models

Now that we actually have a display that behaviour will change e.g. polling of sd-card detect but that's not even beta yet.